ledger/ledgercore/error.go

```diff
@@ -107,3 +107,14 @@ type ErrNonSequentialBlockEval struct {
 func (err ErrNonSequentialBlockEval) Error() string {
 	return fmt.Sprintf("block evaluation for round %d requires sequential evaluation while the latest round is %d", err.EvaluatorRound, err.LatestRound)
 }
+
+// GroupTaskError indicates the group index of the unfulfilled resource
+type GroupTaskError struct {
```

Comment:
two requests:
1. move the GroupTaskError to the prefetcher package.
2. Make GroupTaskError be used with &GroupTaskError, and have the `Unwrap() error` implemented, so that the caller could use the `errors.Is` and `errors.As`.